Extend image_name_template to allow timestamp

Veldhoen opened the pull request:
Fixes #394

Breakthrough commented:
I see in your PR you've already used the `$TIMESTAMP_MS` format.  Would you be able to run `yapf` to format the resulting code so it passes the formatting checks?  Thanks!

Breakthrough reviewed (changes requested):
Thanks for the PR, please address the following so this can be landed:

 - [x] Rebase ontop of the `develop` branch
 - [ ] Ensure the code is formatted correctly: `python -m yapf -i -r scenedetect tests`

Veldhoen commented:
Thanks for replying so fast, even though my PR was still a draft. I'll make sure to address the issues you mentioned, hopefully later today.

Veldhoen commented:
Rebased on `develop` now, and I think the formatting is in line with your standards now. 
Also added `$TIMECODE` like you suggested in the issue, and added both `$TIMECODE` and `$FRAME_NUMBER` to `test_save_images`.

Documentation still needs to be updated to explain these extensions.

Breakthrough commented on scenedetect/scene_manager.py:
```diff
@@ -489,11 +489,16 @@ def save_images(scene_list: List[Tuple[FrameTimecode, FrameTimecode]],
+                        IMAGE_NUMBER=image_num_format % (j + 1),
+                        FRAME_NUMBER=image_timecode.get_frames(),
+                        TIMESTAMP_MS=int(image_timecode.get_seconds() * 1000),
+                        TIMECODE=image_timecode.get_timecode()),
```
Timecodes need to be formatted specially as not all operating systems support `:` in paths.  Please replace them with `;` [as is done when splitting videos](https://github.com/Breakthrough/PySceneDetect/blob/main/scenedetect/video_splitter.py#L140-L141):

```python
TIMECODE=image_timecode.get_timecode().replace(":", ";")
```

This is most likely why `test_save_images` is failing on the Windows build currently.

Breakthrough commented on tests/test_scene_manager.py:
```diff
@@ -94,7 +94,8 @@ def test_save_images(test_video_file):
     sm.add_detector(ContentDetector())
 
     image_name_glob = 'scenedetect.tempfile.*.jpg'
-    image_name_template = 'scenedetect.tempfile.$SCENE_NUMBER.$IMAGE_NUMBER'
```
Could you remove the whitespace present in the resulting string?  You can use string concatenation if you place the expression in brackets:
    
    image_name_template = ('scenedetect.tempfile.'
                           '$SCENE_NUMBER.$IMAGE_NUMBER.$FRAME_NUMBER.$TIMESTAMP_MS.$TIMECODE')

Thank you for updating the unit tests!

Breakthrough reviewed (changes requested):
Looks like the `test_save_images` test is failing on Windows, will approve once fixed.  Thank you!

Veldhoen commented:
I applied the requested changes, and extended docstring. Hope this will do!

Breakthrough commented:
Much appreciated, approved.

Thank you for the PR!

Veldhoen commented:
@Breakthrough When do you plan to release 0.6.4? I created a pre-release of my fork to be able to use the functionality right away, but I'd like to use a proper release at some point, of course.

Breakthrough commented:
I'm hoping to wrap up the next release by the end of this month.  If for whatever reason you need this sooner in production, I can consider pushing a pre-release to PyPI.

Veldhoen commented:
If you can make it by the end of this month, that would be great. Until then we'll be fine with my ad hoc pre-release. Thanks!

Veldhoen commented:
@Breakthrough do you have any updates concerning the release of 0.6.4? Would be much appreciated :)

Breakthrough commented:
All feature work is complete and I'm just finishing up testing and preparing the release.  The new version should be out within a day or two.
